packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -91,17 +91,17 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
         // This way we have latest rows on the first rendering
         const rows = sortedGridRowsSelector(state);
 
-        rows.forEach((row) => {
-          const params = apiRef.current.getCellParams(row.id, filterItem.columnField!);
+        rows.forEach((row: GridRowModel, id: GridRowId) => {
+          const params = apiRef.current.getCellParams(id, filterItem.columnField!);
```

Comment:
On another note, this might be a bit slow 🤔 
Might have to reduce the loaded params for filtering as I have done for sorting.
Need some perf test here